opensearch_py_ml/ml_commons/rag_pipeline/rag/AIConnectorHelper.py

```diff
@@ -0,0 +1,698 @@
+        # Create a boto3 client for IAM
+        iam_client = boto3.client('iam')
+
+        try:
+            # Get information about the IAM user
+            response = iam_client.get_user(UserName=username)
+            user_arn = response['User']['Arn']
+            return user_arn
+        except iam_client.exceptions.NoSuchEntityException:
+            print(f"IAM user '{username}' not found.")
+            return None
+
+    def secret_exists(self, secret_name):
+        secretsmanager = boto3.client('secretsmanager', region_name=self.region)
+        try:
+            # Try to get the secret
+            secretsmanager.get_secret_value(SecretId=secret_name)
+            # If no exception was raised by get_secret_value, the secret exists
+            return True
+        except secretsmanager.exceptions.ResourceNotFoundException:
+            # If a ResourceNotFoundException was raised, the secret does not exist
+            return False
+
+    def get_secret_arn(self, secret_name):
+        secretsmanager = boto3.client('secretsmanager', region_name=self.region)
+        try:
+            response = secretsmanager.describe_secret(SecretId=secret_name)
+            # Return ARN of the secret 
+            return response['ARN']
+        except secretsmanager.exceptions.ResourceNotFoundException:
+            print(f"The requested secret {secret_name} was not found")
+            return None
+        except Exception as e:
+            print(f"An error occurred: {e}")
+            return None
+        
+    def get_secret(self, secret_name):
+        secretsmanager = boto3.client('secretsmanager', region_name=self.region)
+        try:
+            response = secretsmanager.get_secret_value(SecretId=secret_name)
+        except secretsmanager.exceptions.NoSuchEntityException:
+            print("The requested secret was not found")
+            return None
+        except Exception as e:
+            print(f"An error occurred: {e}")
+            return None
+        else:
+            return response.get('SecretString')
+
+    def create_secret(self, secret_name, secret_value):
+        secretsmanager = boto3.client('secretsmanager', region_name=self.region)
+
+        try:
+            response = secretsmanager.create_secret(
+                Name=secret_name,
+                SecretString=json.dumps(secret_value),
+            )
+            print(f'Secret {secret_name} created successfully.')
+            return response['ARN']  # Return the ARN of the created secret
+        except BotoCoreError as e:
+            print(f'Error creating secret: {e}')
+            return None
+
+
+    def role_exists(self, role_name):
+        iam_client = boto3.client('iam')
+
+        try:
+            iam_client.get_role(RoleName=role_name)
+            return True
+        except iam_client.exceptions.NoSuchEntityException:
+            return False
+
+    def delete_role(self, role_name):
+        iam_client = boto3.client('iam')
```

Comment:
rather than every time initializing the iam client, can we have corresponding helper classes like IAMHelper class, secret manager helper class?


Just to give you an example:

```
class IAMHelper:
    def __init__(self):
        self.iam_client = boto3.client('iam')

    def role_exists(self, role_name):
        try:
            self.iam_client.get_role(RoleName=role_name)
            return True
        except self.iam_client.exceptions.NoSuchEntityException:
            return False
```

Reply:
Completed made a separate class for this.